rustls/build.rs

```diff
@@ -0,0 +1,16 @@
+/// This build script allows us to enable the `read_buf` language feature only
+/// for Rust Nightly.
+///
+/// We cannot use `rustversion` to avoid this build script because
+/// `rustversion` doesn't provide a mechanism for conditionally enabling
+/// language features like `read_buf`.
```

Comment:
We can just stick `#[rustversion(nightly)]` on each instance separately and dump the feature, as @ctz suggested? Seems simpler to me.